skfda/preprocessing/dim_reduction/feature_extraction/_fda_feature_union.py

```diff
@@ -0,0 +1,104 @@
+"""Feature extraction union for dimensionality reduction."""
+from __future__ import annotations
+
+from typing import Any, Union
+
+from numpy import ndarray
+from pandas import DataFrame
+from sklearn.pipeline import FeatureUnion
+
+from ....representation.basis import FDataBasis
+from ....representation.grid import FDataGrid
+
+
+class FdaFeatureUnion(FeatureUnion):
+    """Concatenates results of multiple functional transformer objects.
+
+    This estimator applies a list of transformer objects in parallel to the
+    input data, then concatenates the results (They can be either FDataGrid
+    and FDataBasis objects or multivariate data itself).This is useful to
+    combine several feature extraction mechanisms into a single transformer.
+    Parameters of the transformers may be set using its name and the parameter
+    name separated by a '__'. A transformer may be replaced entirely by
+    setting the parameter with its name to another transformer,
+    or removed by setting to 'drop'.
+
+    Parameters:
+        transformer_list: list of tuple
+            List of tuple containing `(str, transformer)`. The first element
+            of the tuple is name affected to the transformer while the
+            second element is a scikit-learn transformer instance.
+            The transformer instance can also be `"drop"` for it to be
+            ignored.
+        n_jobs: int
+            Number of jobs to run in parallel.
+            ``None`` means 1 unless in a :obj:`joblib.parallel_backend`
+            context.
+            ``-1`` means using all processors.
+            The default value is None
+        transformer_weights: dict
+            Multiplicative weights for features per transformer.
+            Keys are transformer names, values the weights.
+            Raises ValueError if key not present in ``transformer_list``.
+        verbose: bool
+            If True, the time elapsed while fitting each transformer will be
+            printed as it is completed. By default the value is False
+        array_output: bool
+            indicates if the transformed data is requested to be a NumPy array
+            output. By default the value is False.
+
+    Examples:
+    Firstly we will import the Berkeley Growth Study data set
+    >>> from skfda.datasets import fetch_growth
+    >>> X = fetch_growth(return_X_y=True)[0]
+
+    Then we need to import the transformers we want to use
+    >>> from skfda.preprocessing.dim_reduction.feature_extraction import FPCA
+    >>> from skfda.representation import EvaluationTransformer
+
+    Finally we import the union and apply fit and transform
+    >>> from skfda.preprocessing.dim_reduction.feature_extraction
+    ... import FdaFeatureUnion
+    >>> union = FdaFeatureUnion([
+    ...    ("Eval", EvaluationTransformer()),
+    ...    ("fpca", FPCA()), ], array_output=True)
+    >>> union.fit_transform(X)
```

Comment:
This test is not passing. Also an example where the new features are concatenated with the original curve would be good.

Reply:
I am wondering how to concatenate the original data to the transformed one as both  have different sizes.
In addition, X.data_matrix is an array of functions, and  each of them an array of array points (dimension 3). The transformed data instead is an array of arrays containing transformed points (dimension 2).

Reply:
You are thinking this wrong. The transformed data always has the same number of samples as the original data. The concatenation is done as separate columns, and all of them have the same number of rows.

Reply:
Okay, I have developed an algorithm that is able to concatenate them, but it is too complex for writing it into the example (it includes two nested loops), should I create it as a function on the module FeatureUnion and call it?